src/components/UserInfoBar/UserInfoBar.tsx

```diff
@@ -32,9 +17,9 @@ export const UserInfoBar = () => {
       </div>
       <div className={classes.userInfoContent}>
         <div className={classes.userInfoTextContainer}>
-          {userInfoName && <p className={classes.userInfoText}>{userInfoName}</p>}
-          {userInfoName !== reporteeName && reporteeName && (
-            <p className={classes.userInfoText}>for {reporteeName}</p>
+          {!userIsFetching && <span className={classes.userInfoText}>{userData.name}</span>}
+          {!reporteeFetching && !userIsFetching && userData?.name !== reporteeData?.name && (
```

Comment:
Looks like we might be missing a condition here? Previously the code checked for `userInfoName !== reporteeName && reporteeName`, while now it checks that the values are not equal. (not sure if it will ever happend that an empty `reporteeData.name` is shown)

Reply:
You're absolutely right. We should only display "for [reporteeData.name]" if reporteeData.name exists and is different from user.name. I'll update the condition to ensure this. Thanks for catching it!